lib/chainsched/chain_sched.go

```diff
@@ -41,7 +48,8 @@ func (s *CurioChainSched) AddHandler(ch UpdateFunc) error {
 	if s.started {
 		return xerrors.Errorf("cannot add handler after start")
 	}
-
+	s.callbacksLk.Lock()
+	defer s.callbacksLk.Unlock()
```

Comment:
I'd put the log above started. Even though I'm 99% sure that all hook adds happen in construction, if we want to use the lock as a guard, for it to actually do anything it has to be above the `if started`